### PR TITLE
 initupload pushes latest-build.txt for every job 

### PR DESCRIPTION
**Pull request by kargakis**
Updates `initupload` to push `latest-build.txt` for every job in
its top-level GCS directory based on its type and the documented
GCS layout.

I am not familiar with the layout that Gubernator expects, let me
know if this is the correct one to follow.

Ref https://github.com/kubernetes/test-infra/issues/7056

/area prow

/cc @stevekuznetsov @rmmh @BenTheElder

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
/lgtm
/hold
Code looks good, any comment on GCS placement @rmmh @BenTheElder ?

File: prow/pod-utils/gcs/target.go [hunk 1]
```diff
@@ -56,6 +56,20 @@ func AliasForSpec(spec *pjutil.JobSpec) string {
...
+// LatestBuildForSpec determines the GCS path for storing the latest
+// build id for a job.
+func LatestBuildForSpec(spec *pjutil.JobSpec) string {
+	switch spec.Type {
+	case kube.PeriodicJob, kube.PostsubmitJob:
+		return path.Join("logs", spec.Job, "latest-build.txt")
+	case kube.PresubmitJob, kube.BatchJob:
+		return path.Join("pr-logs", "directory", spec.Job, "latest-build.txt")
+	default:
+		logrus.Fatalf("unknown job spec type: %v", spec.Type)
```
Comment (rmmh) [inline prow/pod-utils/gcs/target.go]: this should be logrus.Errorf, unless you want to crash the whole process here

Comment (kargakis) [reply prow/pod-utils/gcs/target.go]: Fixed

Comment (kargakis) [reply prow/pod-utils/gcs/target.go]: Although the rest of the helpers use Fatalf for the same cause, ie. can't figure out the pj type so `initupload` will crash anyway.

Comment (kargakis) [reply prow/pod-utils/gcs/target.go]: Since I am going to use this helper in tot I think it's better to avoid crashing.

**Comment by rmmh** [comment]
@kargakis one addition: PR builds upload *two* latest-build.txts: gs://kubernetes-jenkins/pr-logs/pull/59224/pull-kubernetes-typecheck/latest-build.txt and gs://kubernetes-jenkins/pr-logs/directory/pull-kubernetes-typecheck/latest-build.txt

**Comment by kargakis** [comment]
@rmmh opened https://github.com/kubernetes/test-infra/pull/7126

**Pull request merged**